lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -1193,9 +1172,9 @@ static bool canSynthesize(TypeChecker &tc, NominalTypeDecl *target,
                       diag::decodable_super_init_not_designated_here,
                       requirement->getFullName(), memberName);
           return false;
-        } else if (!initializer->isAccessibleFrom(target)) {
+        } else if (!initializer->isAccessibleFrom(classDecl)) {
```

Comment:
I think this should be `initializer->isAccessibleFrom(conformanceDC)`, although I'm not convinced that it will ever matter.